web/handlers.go

```diff
@@ -222,7 +222,7 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // checkCSRFToken performs a CSRF check on the provided request with the given CSRF token. Returns whether or not
 // a CSRF check occurred and whether or not it succeeded.
 func (h *Handler) checkCSRFToken(c *Context, r *http.Request, token string, tokenLocation app.TokenLocation, session *model.Session) (checked bool, passed bool) {
-	csrfCheckNeeded := c.Err == nil && tokenLocation == app.TokenLocationCookie && h.RequireSession && !h.TrustRequester && r.Method != "GET"
+	csrfCheckNeeded := session != nil && c.Err == nil && tokenLocation == app.TokenLocationCookie && !h.TrustRequester && r.Method != "GET"
```

Comment:
Should the `nil` check be moved to line 231 instead? That way requests with `nil` sessions would still get a CSRF check via the legacy `X-Requested-With` approach.

Reply:
Yes, that sounds right, as previously discussed. I am just wondering when the deprecation period for the block below ends.

Reply:
Let's get a comment from @hmhealey or @crspeller on that.

Reply:
We plan to do that deprecation with MMv6.
Do we need to check for CSRF for requests with no session? Seems strange if we plan to remove that functionality later.

Reply:
Might be a safer bet not to rely on the deprecated header.